featurebyte/utils/async_helper.py

```diff
@@ -249,8 +254,6 @@ async def _coro(coro: Coroutine[Any, Any, Any]) -> Any:
                 return
             task = asyncio.create_task(coro)
             tasks.append(task)
-            # Add task to refresh semaphore while task is running
-            semaphore_refresh_task = asyncio.create_task(_semaphore_refresh(task, semaphore))
```

Comment:
Moved handling of the refresh task into the semaphore class so it's self contained